### ISSUE TITLE
Fix FN S6333: 'addMethod' using 'Resource.api.root'

### ISSUE THREAD

**Issue by pedro-oliveira-sonarsource** [SonarSource/SonarJS#3649]
### **What**

In  #3452 , support for _aws_cdk.aws_apigateway.RestApi_ was added, where we raise an issue if an endpoint is defined to be public.

On a validation sprint, when analyzing the [code sample](https://github.com/SonarSource/appsec-poc/blob/master/typescript/poc/rules/S6333/src/stacks/restapi_stack.ts) produced during the research for rule specification a false-negative was detected if _addMethod_ was called from the _[Resource.api.root](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.IRestApi.html#root)_ property.

When  _authorizationType_ is not defined, 'NONE' is the default value.
```
const api = new apigateway.RestApi(this, "RestApi")

const test = api.root.addResource("test")
test.addMethod(
    "GET" // TP - Sensitive
)

test.api.root.addMethod( // FN - Sensitive; no authorization by default
    "DELETE"
)
```
### **How**
When _authorizationType_ is not set, raise an issue using the 'omittedAuthorizationType' messageid.

**Note:** See [issue](https://github.com/SonarSource/SonarJS/issues/3648), for a change of specification to detect the lack of _authorizationType_.

### **Resources**
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.Resource.html
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.Resource.html#api
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.IRestApi.html#root

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-101
